runtime-modules/referendum/src/lib.rs

```diff
@@ -587,56 +548,76 @@ impl<T: Trait<I>, I: Instance> Mutations<T, I> {
                 }
             }
 
-            // option is already in the list and only needs to change it's position?
-            if let Some(current_index) = already_existing_index {
-                return Some(move_winner::<T, I>(
-                    current_winners,
-                    current_index,
-                    insert_index,
-                    option_result.vote_power,
-                ));
-            }
+            let mut new_winners = current_winners.to_vec();
```

Comment:
- This def. looks simpler, I think it could be a bit safer.
  1. `already_existing_index` can be immutable, and in fact can be found using `iter::find`, so no need to do it yourself even.
  2. try a bit more descriptive names. `new_virtual_index` does not help the reader much, even `already_existing_index` could be improved to `current_winners_index_of_vote_recipient`.
  3. The entire block from 554->592 is doing a lot, it computes a value, has side-effects and even has path that exists the method entirely. It takes a lot of time to get convinced that it works. Is it possible to do this in a few more steps that have a bit narrower purpose.

- Can we move out of `reveal_vote`, really obscures readability when an internal utility routine gets this long. Make it a standalone non-public function instead.

Reply:
Code is flattened and simplified.